python/cudf/cudf/core/dataframe.py

```diff
@@ -3916,7 +3916,16 @@ def scatter_by_map(self, map_index, map_size=None, keep_index=True):
                 "Use an integer array/column for better performance."
             )
 
-        tables = self._scatter_to_tables(map_index, keep_index)
+        if map_size is None:
+            map_size = len(map_index.unique())
+        else:
+            if map_size < len(map_index.unique()):
+                raise ValueError(
+                    "ERROR: map_size must be >= %d (got %d)."
+                    % (len(map_index.unique()), map_size)
+                )
```

Comment:
`map_index` is a Column at this point right? If so, couldn't we use `unique_count()` directly?

```suggestion
        if map_size is None:
            map_size = map_index.unique_count()
        else:
            unique_count = map_index.unique_count()
            if map_size < unique_count:
                raise ValueError(
                    "ERROR: map_size must be >= %d (got %d)."
                    % (unique_count, map_size)
                )
```